pkg/handler/enqueue_annotation.go

```diff
@@ -0,0 +1,173 @@
+// resource:
+//
+// ...
+// if err := c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForAnnotation{"Pods.core"}); err != nil {
+//    entryLog.Error(err, "unable to watch Pods")
+//    os.Exit(1)
+// }
+// ...
+//
+// With the annotations:
+//
+// ...
+// annotations:
+//    sigs.k8s.io/primary-resource-name: "my-namespace/my-pod"
+//    sigs.k8s.io/primary-resource-type: "Pods.core"
+// ...
+//
+// The following code will enqueue a Request to the `primary-resource-namespace`, ReplicaSet reconcile,
+// when some change occurs in a ClusterRole resource
+//
+// ...
+// if err := c.Watch(&source.Kind{
+//	  // Watch cluster roles
+//	  Type: &rbacv1.ClusterRole{}},
+//
+//	  // Enqueue ReplicaSet reconcile requests using the
+//	  // namespacedName annotation value in the request.
+//	  &handler.EnqueueRequestForAnnotation{"ReplicaSet.apps"}); err != nil {
+//	      entryLog.Error(err, "unable to watch ClusterRole")
+//	      os.Exit(1)
+//    }
+// }
+// ...
+// With the annotations:
+//
+// ...
+// annotations:
+//    sigs.k8s.io/primary-resource-name: "my-namespace/my-replicaset"
+//    sigs.k8s.io/primary-resource-type: "ReplicaSet.apps"
+// ...
+type EnqueueRequestForAnnotation struct {
+	// It is used to verify that the primary resource is the primary resource to use. It can be any arbitrary value.
+	// However, it would be recommended to fill it with `<Kind>.<group>` from the primary resource to make it easy
+	// to tell which specific resource is interested in.
+	Type string
+}
+
+// Create implements EventHandler
+func (e *EnqueueRequestForAnnotation) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	if ok, req := e.getAnnotationRequests(evt.Meta); ok {
+		q.Add(req)
+	}
+}
+
+// Update implements EventHandler
+func (e *EnqueueRequestForAnnotation) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	if ok, req := e.getAnnotationRequests(evt.MetaOld); ok {
+		q.Add(req)
+	} else if ok, req := e.getAnnotationRequests(evt.MetaNew); ok {
+		q.Add(req)
+	}
+}
+
+// Delete implements EventHandler
+func (e *EnqueueRequestForAnnotation) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	if ok, req := e.getAnnotationRequests(evt.Meta); ok {
+		q.Add(req)
+	}
+}
+
+// Generic implements EventHandler
+func (e *EnqueueRequestForAnnotation) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+	if ok, req := e.getAnnotationRequests(evt.Meta); ok {
+		q.Add(req)
+	}
+}
+
+// getAnnotationRequests will check if the object has the annotations for the watch handler and requeue
+func (e *EnqueueRequestForAnnotation) getAnnotationRequests(object metav1.Object) (bool, reconcile.Request) {
+	if typeString, ok := object.GetAnnotations()[TypeAnnotation]; ok && typeString == e.Type {
+		if typeString, ok := object.GetAnnotations()[TypeAnnotation]; ok && typeString == e.Type {
```

Comment:
Is this line identical to the line above?

Reply:
Ops. mistake. solved